### PR TITLE
Fixed TLS test so it passes on macOS

### PR DESCRIPTION
**Pull request by kozlovic**
Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

**Review by derekcollison — APPROVED** [review]
LGTM

File: server/leafnode_test.go [hunk 1]
```diff
@@ -2624,7 +2624,11 @@ func TestLeafNodeTLSConfigReload(t *testing.T) {
 	// Wait for the error
 	select {
 	case err := <-lg.errCh:
-		if !strings.Contains(err, "unknown") && !strings.Contains(err, "broken") {
+		// Since Go 1.18, we had to regenerate certs to not have to use GODEBUG="x509sha1=1"
+		// But on macOS, with our test CA certs, no SCTs included, it will fail
+		// for the reason "x509: “localhost” certificate is not standards compliant"
+		// instead of "unknown authority".
+		if !strings.Contains(err, "unknown") && !strings.Contains(err, "compliant") {
```
Comment (wallyqs) [inline server/leafnode_test.go:2631]: Looks like this will be fixed in Go 1.20: https://github.com/golang/go/issues/51991

**Review by wallyqs — APPROVED** [review]
LGTM

File: server/leafnode_test.go (hunk 1 above)
Comment (kozlovic) [reply server/leafnode_test.go:2631]: Thanks for looking!

**Pull request merged**